src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaRewriter.cs

```diff
@@ -1121,12 +1122,12 @@ private DebugId GetClosureId(SyntaxNode syntax, ArrayBuilder<ClosureDebugInfo> c
                 closureId = new DebugId(closureDebugInfo.Count, CompilationState.ModuleBuilderOpt.CurrentGenerationOrdinal);
             }
 
-            int syntaxOffset = _topLevelMethod.CalculateLocalSyntaxOffset(syntax.SpanStart, syntax.SyntaxTree);
+            int syntaxOffset = _topLevelMethod.CalculateLocalSyntaxOffset(syntax.SpanStart, syntax.SyntaxTree, localKind);
```

Comment:
Given that none of us seem to completely understand the circumstances in which these methods get called, we might be better off being permissive rather than trying to restrict the allowed circumstances to the narrowest possible. If the filter is too restrictive, the symptom will be that the compiler crashes when instrumenting some projects. If the filter is too permissive, the symptom will be that some hypothetical internal compiler errors might not crash the compiler.